x-pack/plugins/ml/server/shared_services/errors.ts

```diff
@@ -5,9 +5,16 @@
  * 2.0.
  */
 
-export class MLClusterClientUninitialized extends Error {
-  constructor(message?: string) {
-    super(message);
-    Object.setPrototypeOf(this, new.target.prototype);
-  }
-}
+export const getCustomError = (className: string, errorMessage: string) => {
```

Comment:
the typical way to check custom errors is with `instanceof`, I believe errors produced by this factory will not allow this.
I appreciate you could compare the error name, but that is not common practice as far as i know.

This is also probably a bit over engineered for what we need here, all we really need is a single custom error, something like `MLShareServcesDependencyError` and the description will differentiate the three types we have.